server/src/main/java/com/linecorp/centraldogma/server/internal/api/AdministrativeService.java

```diff
@@ -124,6 +124,28 @@ public ServerStatus status() {
                 }
                 return status();
             });
+        } else if (oldStatus.replicating) {
+            if (!replicating) { // replicating -> unreplicating
+                logger.warn("Disabling replication with read-only mode");
+                return executor().stop().handle((unused, cause) -> {
+                    if (cause != null) {
+                        logger.warn("Failed to stop the command executor:", cause);
+                    } else {
+                        logger.info("Disabled replication");
+                    }
+                    return status();
+                });
+            }
+        } else if (replicating) { // unreplicating -> replicating
+            logger.warn("Enabling replication with read-only mode");
+            return executor().start().handle((unused, cause) -> {
+                if (cause != null) {
+                    logger.warn("Failed to start the command executor:", cause);
+                } else {
+                    logger.info("Enabled replication");
+                }
+                return status();
+            });
```

Comment:
Could we simplify this if-else chains? For example:
```java
if (oldStatus.writable != writable) {
     executor().setWritable(writable);
}
if (oldStatus.replicating != replicating) {
   if (replicating) {
        // log
        executor().start().handle(...);
   } else {
        // log
        executor().stop().handle(...);
   }
}
```

Reply:
Otherwise, extract common code to template methods?

Reply:
I added tailing else-if because of the more specific logging messages (such as _with replication enabled_ or with _read-only mode_). I agree that we need simplification. If we can give up the messages, let me simplify.

> extract common code to template methods?

May I ask what _template methods_ means?